lib/backend/k8s/resources/workloadendpoint.go

```diff
@@ -171,22 +161,25 @@ func (c *KubeClient) listWorkloadEndpoints(ctx context.Context, l model.Resource
 	}, nil
 }
 
-// getWorkloadEndpoint gets the WorkloadEndpoint from the k8s API based on existing Pods.
-func (c *KubeClient) getWorkloadEndpoint(ctx context.Context, k model.ResourceKey, revision string) (*model.KVPair, error) {
-	// The workloadID is of the form namespace.podname.  Parse it so we
-	// can find the correct namespace to get the pod.
-	namespace, podName := c.converter.ParseWorkloadID(k.Name)
+func (c *WorkloadEndpointClient) EnsureInitialized() error {
+	return nil
+}
 
-	pod, err := c.clientSet.Pods(namespace).Get(podName, metav1.GetOptions{})
-	if err != nil {
-		return nil, resources.K8sErrorToCalico(err, k)
+func (c *WorkloadEndpointClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
+	if len(list.(model.ResourceListOptions).Name) != 0 {
+		return nil, fmt.Errorf("cannot watch specific resource instance: %s", list.(model.ResourceListOptions).Name)
 	}
 
-	// Decide if this pod should be displayed.
-	if !c.converter.IsReadyCalicoPod(pod) {
-		return nil, cerrors.ErrorResourceDoesNotExist{Identifier: k}
+	k8sWatch, err := c.clientSet.CoreV1().Pods("").Watch(metav1.ListOptions{ResourceVersion: revision})
```

Comment:
You need to pass in the namespace from the ListOptions here.  (granted it will be blank for the Syncer use case)